web3/content_script.js

```diff
@@ -155,23 +155,41 @@ function main () {
     // lazy load styles
     const styles = require('./styles.css');
     const iframe = document.createElement('iframe');
-    iframe.className = styles.iframe__main;
+    iframe.className = [ styles.iframe__main, styles.iframe__open ].join(' ');
     iframe.src = chrome.extension.getURL('web3/embed.html');
     iframeInjected = iframe;
 
     window.addEventListener('message', (ev) => {
       if (ev.source !== iframe.contentWindow) {
         return;
       }
-      if (!ev.data.type || ev.data.type !== EV_SIGNER_BAR) {
+
+      if (!ev.data.type) {
         return;
       }
-      if (ev.data.opened) {
-        iframe.classList.add(styles.iframe__open);
-      } else {
-        iframe.classList.remove(styles.iframe__open);
+
+      switch (ev.data.type) {
```

Comment:
TBH I prefer a series of early-exiting ifs instead of `switch` (less identation, clearer flow, no risk of forgeting to break/return):
```js
if (type === EV_SIGNER_BAR) {
  // do something
  return;
}

if (type === EV_IFRAME_STYLE) {
  // do something
  return;
}
```
but I'm also ok with this approach if you give me one argument why you prefer it :)